templates/zerver/search_operators.html

```diff
@@ -22,6 +22,10 @@
                 <td class="definition">{{ _('Narrow to private messages with') }} <span class="operator_value">email</span></td>
             </tr>
             <tr>
+                <td class="operator">huddle-with:<span class="operator_value">email</span></td>
+                <td class="definition">{{ _('Narrow to group private messages with') }} <span class="operator_value">email</span></td>
+            </tr>
+            <tr>
```

Comment:
You should also document this is `templates/zerver/help/search-for-messages.html`, with a more detailed description that explains things like that this only searches for group PMs (not 1:1 PMs) that contain both you and the other user.